src/pytorch_lightning/_graveyard/core.py

```diff
@@ -13,25 +13,47 @@
 # limitations under the License.
 from typing import Any
 
-from pytorch_lightning import LightningDataModule
+from pytorch_lightning import LightningDataModule, LightningModule
 
 
 def _on_save_checkpoint(_: LightningDataModule, __: Any) -> None:
     # TODO: Remove in v2.0.0
     raise NotImplementedError(
-        "`LightningDataModule.on_save_checkpoint` was deprecated in v1.6 and removed in v1.8."
+        "`LightningDataModule.on_save_checkpoint` was deprecated in v1.6 and is no longer supported in v1.8."
         " Use `state_dict` instead."
     )
 
 
 def _on_load_checkpoint(_: LightningDataModule, __: Any) -> None:
     # TODO: Remove in v2.0.0
     raise NotImplementedError(
-        "`LightningDataModule.on_load_checkpoint` was deprecated in v1.6 and removed in v1.8."
+        "`LightningDataModule.on_load_checkpoint` was deprecated in v1.6 and is no longer supported in v1.8."
         " Use `load_state_dict` instead."
     )
 
 
+def _use_amp(_: LightningModule) -> None:
+    # Remove in v2.0.0 and the skip in `__jit_unused_properties__`
+    if not LightningModule._jit_is_scripting:
+        # cannot use `AttributeError` as it messes up with `nn.Module.__getattr__`
+        raise RuntimeError(
```

Comment:
I found this too and tried many different ways https://github.com/Lightning-AI/lightning/pull/15059/commits/14a1969e7e0cf5e8ad15da35b78593999ad83637 but never worked. But of course if @carmocca touches the code it instantly works...

Reply:
yeah, I saw your commit messages so that's why I did it here haha

You were just blinded by frustration. It's alright